firedrake/cython/dmcommon.pyx

```diff
@@ -3823,13 +3976,16 @@ def submesh_update_facet_labels(PETSc.DM dm, PETSc.DM subdm):
         PETSc.DMLabel sub_int_facet_label, sub_ext_facet_label
         PetscBool has_point
 
+    dim = dm.getDimension()
+    subdim = subdm.getDimension()
+    if subdim != dim:
+        # TODO: Do something useful if necessary.
```

Comment:
Please can you add more detail here describing when you might hit this condition and what the useful something might be?

Reply:
We hit this whenever `subdim != dim`, e.g. when making a co-dim 1 submesh. We currently do nothing in such case. I added a one-line comment to explain more.